lib/hutch/publisher.rb

```diff
@@ -27,30 +27,29 @@ def publish(routing_key, message, properties = {}, options = {})
       properties[:message_id]   ||= generate_id
 
       payload = serializer.encode(message)
-
-      log_publication(serializer, payload, routing_key)
-
-      response = exchange.publish(payload, {persistent: true}.
+      payload_properties = {persistent: true}.
         merge(properties).
         merge(global_properties).
-        merge(non_overridable_properties))
+        merge(non_overridable_properties)
+
+      log_publication(serializer, payload, payload_properties, routing_key)
+      response = exchange.publish(payload, payload_properties)
 
       channel.wait_for_confirms if config[:force_publisher_confirms]
       response
     end
 
     private
 
-    def log_publication(serializer, payload, routing_key)
-      logger.info {
-        spec =
-          if serializer.binary?
-            "#{payload.bytesize} bytes message"
-          else
-            "message '#{payload}'"
-          end
-        "publishing #{spec} to #{routing_key}"
-      }
+    def log_publication(serializer, payload, payload_properties, routing_key)
+      logger.info("publishing message with id: #{payload_properties[:message_id]} to #{routing_key}")
```

Comment:
_🙋‍♂generic observation_

It seems `routing_key` is included in `payload_properties` due to the `non_overridable_properties` above.

What do you think about using `def log_publication(serializer, payload, message_id, routing_key)` as the signature to be more specific and avoid passing down the `routing_key` twice?

Reply:
Sounds good @sldblog!